### DISCUSSION TITLE
Allow PDFDocumentBuilder to save PDF into a stream

### DISCUSSION THREAD

**Discussion by ghost** [UglyToad/PdfPig#562] [Ideas]
For now you can not get PDF with a size more than 2 gigs. And it takes a lot of memory to obtain the raw bytes in the end.
I suggest instead of Build method to add a SaveTo method to save the results into a stream. Which can be a MemoryStream, a FileStream, etc ...
```      
        public void SaveTo(Stream destination)
        {
            CompleteDocument();

            if (!context.Stream.CanSeek)
            {
                throw new InvalidOperationException("PdfDocument.Build() called with non-seekable stream.");
            }

            context.Stream.Seek(0, SeekOrigin.Begin);
            context.Stream.CopyTo(destination);
        }
```

The Build method is better to rename to something like ToByteArray, because it does not do anything except converting the result stream to bytes.

**Comment by cremor** [comment]
`PdfDocumentBuilder` offers a constructor overload that accepts a `Stream` parameter. There you can pass your target stream.
`PdfDocumentBuilder.Dispose()` will call `CompleteDocument()`. So just put your builder in a `using` statement and you will have a finished PDF after the `using` block in your target stream without having to build any byte array.

I agree that the API is not very nice since you need to know what `PdfDocumentBuilder.Dispose()` does, but at least it should work.
